### DISCUSSION TITLE
Zig程序设计语言中文手册

### DISCUSSION THREAD

**Discussion by sxwangzhiwen** [zigcc/forum#36] [General]
我写了Zig程序设计语言中文手册，参考了官方英文手册、标准库和语言源代码，但不是官方手册的全文翻译。其中绝大部分示例程序都实际运行测试过。

章节是主要分为以下几大块：变量、类型、指针、语句等语法结构；类型转换、反射、未定义行为等常用功能；编译期计算、异步函数、编译、测试等Zig特色功能；内存分配器、文件读写、格式化输入输出等常用函数。

手册的网址是：
https://sxwangzhiwen.github.io/zigcndoc/zigcndoc.html

**Comment by sxwangzhiwen** [comment]
并且用Zig语言实际开发了1个格式转换单机程序（源代码3000行左右），这个html手册就是用这个小工具格式转换生成的。https://github.com/sxwangzhiwen/progdoc

**Reply by jiacai2050** [reply]
赞👍，欢迎大佬加入我们的微信群

**Reply by hhstore** [reply]
> 并且用Zig语言实际开发了1个格式转换单机程序（源代码3000行左右），这个html手册就是用这个小工具格式转换生成的。https://github.com/sxwangzhiwen/progdoc

- 看了一下源码, 是中文函数名和变量名. 
- 你这加密方式, 是怕别人(非华语圈)看懂代码吗? 
- 尝试阅读了一下, 头疼. 佩服你能坚持写完+编译通过. 
- 写的很好, 下次别这样了. 😄

**Reply by jiacai2050** [reply]
觉得用中文是 feature，不是 bug

**Reply by sxwangzhiwen** [reply]
我在注释里写了，编程最难的是架构设计，最痛苦的是起名字和中后期DEBUG。
我做为一个英文不好的编程爱好者，用中文函数名和变量名效率提升数倍，再也不用痛苦的找合适英文名字了，而且可以少写注释。

另外本来就是个小工具，我都不指望中国这么多程序员参与，何况是外国人呢。何况绝大多数中小项目，参与写代码的人本来就没几个。

写这个程序就是出于我自己写手册转换格式的需要，我想怎么写就怎么写，与你有什么关系呢？你可以看也可以不看。
你要是个外国人看不懂中文说几句怪话还可以理解，看你说的中文挺标准的，为什么看中文代码头疼呢？
不需要你居高临下的建议，也请你下次别这样指指点点了。

**Reply by sxwangzhiwen** [reply]
有点儿跑题。

我写Zig程序设计语言中文手册纯是出于个人兴趣，起初只是记录自己学习过程中的一些心得，后来觉得英文手册查起来费劲，而且章节编排不适合我自己的感觉，所以才自己写的，想着便于以后写代码时方便查阅。

写文档的时候，不想用复杂的文档格式，就一边写中文手册，一边构思。用了1个月写中文手册，写完后用1个礼拜设计程序，2个礼拜开发写代码。

我在官网上找了找，Zig语言中文社区好像就这一个。我知道我写的中文手册里肯定有错漏的内容，程序也有BUG。大家觉得有用，就看看，没用就拉倒。互联网上这么多内容，觉得不合适可以看别的去。我弄这个事儿纯属自己的兴趣爱好，我又不靠这个吃饭挣钱，我发到这里只是想着帮助提升一下Zig语言在中文程序员中的使用程度，没想着让大家评判。

**Reply by jiacai2050** [reply]
嗯，有实际产出总是好的，开源就是这样， 先有个粗糙的一个版本，然后社区再逐渐完善，不可能指望一个人搞的多完美。

还是为你的行动点赞。

**Reply by hhstore** [reply]
- 这么敏感? 
- 我寻思, 我也没说啥? 
- 你用啥方式/风格写, 是你的自由. 
- 你不开源, 也没人 care.
- 开源了, 又给他人阅读+理解, 制造障碍. 
- 这心态很奇怪. 
- 喜欢这样写, 就多这样写. 加油.  
- 无需再回. 开心就好.

**Comment by sxwangzhiwen** [comment]
好奇怪，用中文给你阅读理解制造障碍了？
